src/projects/gt4o_us/quick-arcade/ArcadeRoot.ad

```diff
@@ -0,0 +1,533 @@
+		widget.actor.warp();
+		widget.actor.out = 0;
+	}
+
+	function rewindFadeActor(widget)
+	{
+		widget.actor.out = 1;
+		widget.actor.warp();
+		widget.actor.out = 0;
+	}
+
+	function setScrollActor(widget)
+	{
+		widget.actor = main::menu::MMoveActor(widget, 0.0, 0.0);
+		widget.actor.ratio = 0.2;
+		widget.actor.out = 1;
+		widget.actor.warp();
+	}
+
+	function setMoveActor(widget, x, y, r)
+	{
+		var actor = main::menu::MMoveActor(widget, widget.x + x, widget.y + y);
+		widget.actor = actor;
+		widget.actor.ratio = r;
+		widget.actor.out = 1;
+		widget.actor.warp();
+	}
+
+	function rewindMoveActor(widget)
+	{
+		widget.actor.out = 1;
+		widget.actor.warp();
+		widget.actor.out = 0;
+	}
+
+	function getIconTopIndex(context)
+	{
+		var num = icon_name_tbl.size;
+		var i = 0;
+	
+		while (i < num)
+		{
+			if (ArcadeRoot::Buttons[icon_name_tbl[i]].visible == 1)
+			{
+				return i;
+			}
+	
+			i++;
+		}
+	
+		return -1;
+	}
+
+	function setDefaultIconFocus(context)
+	{
+		var quick_work = main::menu::MQuickWork();
+		var idx = quick_work.cursorPosition;
+	
+		if (ArcadeRoot::Buttons[icon_name_tbl[idx]].visible != 1)
+		{
+			idx = getIconTopIndex(context);
+		}
+	
+		ArcadeRoot.setFocus(Buttons[icon_name_tbl[idx]]);
+	}
+
+	function getIconIndex(context, icon_name)
+	{
+		var idx = -1;
+		var num = icon_name_tbl.size;
+	
+		for (var i = 0; i < num; i++)
+		{
+			if (icon_name_tbl[i] == icon_name)
+			{
+				idx = i;
+				break;
+			}
+		}
+	
+		return idx;
+	}
+	// end icon.ad included stuff
+	
+	//#include "../quick-share/quick_tune.ad";
+	// Impractical include
+	// begin quick_tune.ad included stuff
+
+	static tirename_tbl = [
+	"tire_short_name_Economy",
+	"tire_short_name_Comfort",
+	"tire_short_name_Street",
+    	"tire_short_name_SprortsHard",
+    	"tire_short_name_SprortsMedium",
+    	"tire_short_name_SprortsSoft",
+    	"tire_short_name_RacingSuperHard",
+    	"tire_short_name_RacingHard",
+    	"tire_short_name_RacingMedium",
+    	"tire_short_name_RacingSoft",
+    	"tire_short_name_RacingSuperSoft",
+   	"tire_short_name_Dirt",
+    	"tire_short_name_Snow"
+	];
+
+	// end quick_tune.ad included stuff
+
+	function updateMyCarInfo(context)
+	{
+		var quick_work = main::menu::MQuickWork();
+		var i = quick_work.numberOfEntries - 1;
+	
+		if (quick_work.getPower(i) == 0)
+		{
+			ArcadeRoot["Grid" + (i + 1)]["PS"].text = "--";
+		}
+		else
+		{
+			var ps = (quick_work.getPower(i) * (100 + quick_work.QuickTunePowerLevel)) / 100;
+			ArcadeRoot["Grid" + (i + 1)]["PS"].text = main::menu::MUnit::GetPowerForDisplay(ps);
+		}
+		ArcadeRoot["Grid" + (i + 1)]["KG"].text = (quick_work.getWeight(i) * (100 + quick_work.QuickTuneWeightLevel)) / 100;
+		setTireText(context, i, quick_work.QuickTuneFrontTireType);
+	}
+
+	function setTireText(context, grid, type)
+	{
+		if (type >= tirename_tbl.size || type < 0)
+		{
+			if (type != 255)
+			{
+				ArcadeRoot["Grid" + (grid + 1)]["Tire"].text = "--";
+			}
+			return;

```

Comment:
```
2A77|  78| 10| VARIABLE_EVAL: type, Local:2
2A8A|  78| 11| INT_CONST: 255 (0xFF)
2A93|  78| 12| BINARY_OPERATOR: != (!=)
2A9C|  80| 13| JUMP_IF_FALSE: Jump To Func Ins 15
  2AA5|  80| 14| JUMP: JumpTo=15
2AAE|  80| 15| VARIABLE_EVAL: ArcadeRoot,ArcadeRoot, Static:5
...
```

type check is in an empty block. There's no return either, instead it jumps to after the next block